engine/engine.go

```diff
@@ -173,6 +173,10 @@ func New(opts Opts) *compatibilityEngine {
 		opts.ExtLookbackDelta = 1 * time.Hour
 		level.Debug(opts.Logger).Log("msg", "externallookback delta is zero, setting to default value", "value", 1*24*time.Hour)
 	}
+	if opts.MaxSamples == 0 {
+		opts.MaxSamples = math.MaxInt
```

Comment:
I suggest not doing that. See other comment.

Reply:
done